include/migration/migration.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MIGRATION_HPP_
+#define MIGRATION_HPP_
+
+#include <Poco/Data/Session.h>
+
+class Migration {
+public:
+    /**
+     * Creates database migrator instance.
+     *
+     * @param session database session.
+     */
+    explicit Migration(Poco::Data::Session& session, const std::string& migrationDir);
+
+    /**
+     * Migrates database to the specified version.
+     *
+     * @param targetVersion target database version.
+     */
+    void MigrateToVersion(int targetVersion);
+
+    /**
+     * Returns current database version.
+     *
+     * @return current database version.
+     */
+    int GetCurrentVersion();
+
+private:
+    void ApplyMigration(const std::string& migrationScript);
+    void CreateVersionTableIfNeeded();
+    void UpdateVersion(int version);
+    void UpgradeDatabase(int targetVersion, int currentVersion);
+    void DowngradeDatabase(int targetVersion, int currentVersion);
+
+    Poco::Data::Session& session_;
+    const std::string    migrationDir_;
+};
+
+#endif /* MIGRATION_HPP_ */
```

Comment:
Please add empy line at the end of the file

Reply:
Done